### ISSUE TITLE
LearningShapelets: impossible to pickle the model

### ISSUE THREAD

**Issue by alevangel** [tslearn-team/tslearn#403]
https://github.com/tslearn-team/tslearn/blob/42a56cce63d8263982d616fc2bef9009ccbedab4/tslearn/shapelets/shapelets.py#L212-L886

I'm having issues trying to save my fitted model, getting this error:
`AttributeError: Can't pickle local object 'make_gradient_clipnorm_fn.<locals>.<lambda>'`

How to handle it?

**To reproduce:**
```
from tslearn.shapelets import LearningShapelets

clf = LearningShapelets(n_shapelets_per_size={20:1, 30:1},
                                max_iter=1000,
                                optimizer=Adam(learning_rate=0.1),
                                verbose=0)
clf.fit(X_train, y_train)
clf.to_pickle("path/to/file.pkl")
```

**Versions:**
python 3.6
tslearn 0.5.2

**Comment by GillesVandewiele** [comment]
Does the fix described in #387 resolve the issue?

**Comment by alevangel** [comment]
> Does the fix described in #387 resolve the issue?

@GillesVandewiele Nope, because this tryies to solve the loading pointing out that "from_" is a method class, but I don't see nothing about saving the model different from what I did. :(

**Comment by GillesVandewiele** [comment]
It could perhaps be related to an older version of tensorflow, or something related to the optimizer being used: https://github.com/NREL/phygnn/issues/16

One way to circumvent the issues is to just store the learned shapelets and load them in again when doing inference, these contain all the valuable information of a shapelet model.

**Comment by alevangel** [comment]
@

> It could perhaps be related to an older version of tensorflow, or something related to the optimizer being used: [NREL/phygnn#16](https://github.com/NREL/phygnn/issues/16)
> 
> One way to circumvent the issues is to just store the learned shapelets and load them in again when doing inference, these contain all the valuable information of a shapelet model.

Yes, this is absolutely the way of hacking this issue. 

For the version of tensorflow I'm using is 2.6.2 (updated the original post)

**Comment by rtavenar** [comment]
@alevangel : could you post a code snippet that does that?
This would be helpful:
* for other users to see how to handle such errors
* in the future for `tslearn` devs (maybe you?) to figure out how to implement a fix

**Comment by alevangel** [comment]
I undesrtood that the problem is for the Adam optimizer, that have problem to be serialized. I'm trying to understand how to fix it.

@rtavenar sure, I'll share it here as soon as I have it.
